src/xbot/common/command/BaseRobot.java

```diff
@@ -180,14 +183,18 @@ protected void sharedPeriodic() {
         double timeSinceLastLog = Timer.getFPGATimestamp() - lastFreqCounterResetTime;
         if(lastFreqCounterResetTime <= 0) {
             lastFreqCounterResetTime = Timer.getFPGATimestamp();
+            log.info("Total Current:" + pdp.getTotalCurrent());
         }
         else if(timeSinceLastLog >= frequencyReportInterval.get()) {
             double loopsPerSecond = loopCycleCounter / timeSinceLastLog; 
+            double loopDuration = timeSinceLastLog / loopCycleCounter;
```

Comment:
Noooooooo this is just the inverse!

You gotta actually keep track of two times - when the loop started, and when the loop finished.